kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -0,0 +1,554 @@
+# !/bin/sh
+# Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+#
+
+#
+# Function to get server start policy at cluster level
+# $1 - Domain resource in json format
+# $2 - Name of cluster
+# $3 - Return value containing cluster level server start policy
+#
+function getClusterPolicy {
+  local domainJson=$1
+  local clusterName=$2
+  local __clusterPolicy=$3
+  local effectivePolicy=""
+
+  clusterPolicyCmd="(.spec.clusters[] \
+    | select (.clusterName == \"${clusterName}\")).serverStartPolicy"
+  effectivePolicy=$(echo ${domainJson} | jq "${clusterPolicyCmd}")
+  eval $__clusterPolicy=${effectivePolicy}
+}
+
+#
+# Function to get server start policy at domain level
+# $1 - Domain resource in json format
+# $2 - Return value containing domain level server start policy
+#
+function getDomainPolicy {
+  local domainJson=$1
+  local __domainPolicy=$2
+  local effectivePolicy=""
+
+  clusterPolicyCmd=".spec.serverStartPolicy"
+  effectivePolicy=$(echo ${domainJson} | jq "${clusterPolicyCmd}")
+  eval $__domainPolicy=${effectivePolicy}
+}
+
+#
+# Function to get effective start policy of server
+# $1 - Domain resource in json format
+# $2 - Name of server
+# $3 - Name of cluster
+# $4 - Return value containing effective server start policy
+#
+function getEffectivePolicy {
+  local domainJson=$1
+  local serverName=$2
+  local clusterName=$3
+  local __currentPolicy=$4
+  local policyFound=false
+  local managedServers=""
+
+  # Get server start policy for this server
+  managedServers=$(echo ${domainJson} | jq -cr '(.spec.managedServers)')
+  if [ "${managedServers}" != "null" ]; then
+    extractPolicyCmd="(.spec.managedServers[] \
+      | select (.serverName == \"${serverName}\") | .serverStartPolicy)"
+    currentPolicy=$(echo ${domainJson} | jq -r "${extractPolicyCmd}")
+    if [[ -n ${currentPolicy} && ${currentPolicy} != "null" ]]; then
+      # Start policy is set at server level, return policy
+      eval $__currentPolicy="'${currentPolicy}'"
+      policyFound=true
+    fi
+  fi
+  if [ "${policyFound}" == 'false' ]; then
+    clusterPolicyCmd="(.spec.clusters[] \
+      | select (.clusterName == \"${clusterName}\")).serverStartPolicy"
+    currentPolicy=$(echo ${domainJson} | jq -r "${clusterPolicyCmd}")
+    if [ "${currentPolicy}" == "null" ]; then
+      # Start policy is not set at cluster level, check at domain level
+      clusterPolicyCmd=".spec.serverStartPolicy"
+      currentPolicy=$(echo ${domainJson} | jq -r "${clusterPolicyCmd}")
+      if [ "${currentPolicy}" == "null" ]; then
+        # Start policy is not set at domain level, default to IF_NEEDED
+        currentPolicy=IF_NEEDED
+      fi
+    fi
+    eval $__currentPolicy="'${currentPolicy}'"
+  fi
```

Comment:
Can this be simplified  to call 'getServer/Cluster/DomainPolicy' instead of calling the same code as those three functions?